src/Team-Services-Bot.Api/Extensions/BotDataBagExtensions.cs

```diff
@@ -30,14 +32,72 @@ public static class BotDataBagExtensions
         /// <returns>A string representing the account.</returns>
         public static string GetCurrentAccount(this IBotDataBag dataBag)
         {
-            string result;
+            if (dataBag == null)
+            {
+                throw new ArgumentNullException(nameof(dataBag));
+            }
+
+            return dataBag.TryGetValue(Account, out string result) ? result : string.Empty;
+        }
 
+        /// <summary>
+        /// Gets the current profile.
+        /// </summary>
+        /// <param name="dataBag">The <see cref="IBotDataBag"/>.</param>
+        /// <returns>A string representing the account.</returns>
+        public static VstsProfile GetCurrentProfile(this IBotDataBag dataBag)
+        {
             if (dataBag == null)
             {
                 throw new ArgumentNullException(nameof(dataBag));
             }
 
-            return dataBag.TryGetValue(Account, out result) ? result : string.Empty;
+            return dataBag.TryGetValue(Profile, out VstsProfile profile) ? profile : null;
+        }
+
+        /// <summary>
+        /// Gets the current team project for the user.
+        /// </summary>
+        /// <param name="dataBag">The <see cref="IBotDataBag"/>.</param>
+        /// <returns>the name of the current team project.</returns>
+        public static string GetCurrentTeamProject(this IBotDataBag dataBag)
+        {
+            if (dataBag == null)
+            {
+                throw new ArgumentNullException(nameof(dataBag));
+            }
+
+            return dataBag.TryGetValue(TeamProject, out string teamProject) ? teamProject : string.Empty;
+        }
+
+        /// <summary>
+        /// Gets the pin.
```

Comment:
Planning to do better documentation after we have some code in place. So we can build up a common way of writing comments.

But for now, when the user has authorized the bot to access their profile. The authorize page shows a pin that needs to be entered in the chat app. 
This is kind of a re-entry point, so we can track where the user is with the flow.